File: app/src/main/java/ch/bailu/aat/services/location/RealLocationInformation.java
package ch.bailu.aat.services.location;

import android.location.Location;

import ch.bailu.util_java.foc.Foc;
import ch.bailu.util_java.foc.FocName;

public class RealLocationInformation extends LocationInformation {
    private final Location location;
    private final int state;
    private final FocName provider;

    public RealLocationInformation(Location l, int s) {
        provider = new FocName(l.getProvider());
        state = s;
        location = l;
    }

    @Override
    public int getState() {
        return state;
    }

    @Override
    public Foc getFile() {
        return provider;
    }

    @Override
    public float getAccuracy() {
        return location.getAccuracy();
    }

    @Override
    public float getSpeed() {
        return location.getSpeed();
    }

    @Override
    public short getAltitude() {
        return (short)Math.round(location.getAltitude());
    }

    @Override
    public double getLatitude() {
        return location.getLatitude();
    }

    @Override
    public double getLongitude() {
        return location.getLongitude();
    }

    @Override
    public long getTimeStamp() {
        return location.getTime();
    }

    @Override
    public int getLatitudeE6() {
        return (int)(getLatitude()*1e6d);
    }

    @Override
    public int getLongitudeE6() {
        return (int)(getLongitude()*1e6d);
    }

    @Override
    public boolean hasAccuracy() {
        return location.hasAccuracy();
    }

    @Override
    public boolean hasSpeed() {
        return location.hasSpeed();
    }

    @Override
    public boolean hasAltitude() {
        return location.hasAltitude();
    }

    @Override
    public boolean hasBearing() {
        return location.hasBearing();
    }

    @Override
    public boolean isFromGPS() {
        return false;
    }

    @Override
    public long getCreationTime() {
        return getTimeStamp();
    }

    @Override
    public void setAltitude(double altitude) {
        location.setAltitude(altitude);
    }
}

File: aat-android/src/main/java/ch/bailu/aat/services/location/RealLocationInformation.kt
package ch.bailu.aat.services.location

import android.location.Location
import ch.bailu.aat_lib.service.location.LocationInformation
import ch.bailu.foc.Foc
import ch.bailu.foc.FocName

open class RealLocationInformation(private val location: Location, private val state: Int) : LocationInformation() {
    private val provider: FocName = FocName(location.provider)

    override fun getState(): Int {
        return state
    }

    override fun getFile(): Foc {
        return provider
    }

    override fun getAccuracy(): Float {
        return location.accuracy
    }

    override fun getSpeed(): Float {
        return location.speed
    }

    override fun getAltitude(): Double {
        return location.altitude
    }

    override fun getLatitude(): Double {
        return location.latitude
    }

    override fun getLongitude(): Double {
        return location.longitude
    }

    override fun getTimeStamp(): Long {
        return location.time
    }

    override fun getLatitudeE6(): Int {
        return (latitude * 1e6).toInt()
    }

    override fun getLongitudeE6(): Int {
        return (longitude * 1e6).toInt()
    }

    override fun hasAccuracy(): Boolean {
        return location.hasAccuracy()
    }

    override fun hasSpeed(): Boolean {
        return location.hasSpeed()
    }

    override fun hasAltitude(): Boolean {
        return location.hasAltitude()
    }

    override fun hasBearing(): Boolean {
        return location.hasBearing()
    }

    override fun isFromGPS(): Boolean {
        return false
    }

    override fun getCreationTime(): Long {
        return timeStamp
    }

    override fun setAltitude(altitude: Double) {
        location.altitude = altitude
    }
}
